app/models/sample.rb

```diff
@@ -520,6 +522,13 @@ def check_status
     end
   end
 
+  # Delay determined based on query of historical upload times, where 80%
+  # of successful uploads took less than 3 hours by client_updated_at.
+  def self.stalled_uploads(delay = 3.hours)
+    where(status: STATUS_CREATED)
```

Comment:
Sorry to be adding more comments here, but in order to make sure we're only dealing with local uploads, we also need to check that `sample.input_files.pluck(:source_type)` are both set to "local". 

Context: We often notice lots of S3 uploads waiting for a resque worker (this will happen more often once people upload from Basespace), and it would be wrong to fail those samples, regardless of how long they have to wait for their upload to come through.

Reply:
That's good context, and I will restrict to local files, but wouldn't we want an alert if an s3 upload takes longer than 3 hours?

Reply:
> That's good context, and I will restrict to local files, but wouldn't we want an alert if an s3 upload takes longer than 3 hours?

We're specifically trying to detect cases where user gave up or their network got interrupted, right? In case of S3 or basespace upload, this doesn't really happen. It would be either just waiting in the queue or it's a really huge file that just takes a long time to transfer but will complete eventually. Maybe we don't want to mix all these different types of concerns.